File: src/main/java/ca/yyx/hu/BootCompleteReceiver.java
package ca.yyx.hu;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Handler;

import ca.yyx.hu.location.GpsLocationService;

/**
 * @author algavris
 * @date 18/12/2016.
 */
public class BootCompleteReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(final Context context, Intent intent) {

        Handler h = new Handler();
        h.postDelayed(new Runnable() {
            @Override
            public void run() {
                App.get(context).startService(GpsLocationService.intent(context));
            }
        }, 10000);
    }
}

File: app/src/main/java/ca/anodsplace/headunit/app/BootCompleteReceiver.kt
package ca.anodsplace.headunit.app

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Handler
import ca.anodsplace.headunit.App

import ca.anodsplace.headunit.location.GpsLocationService

/**
 * @author algavris
 * *
 * @date 18/12/2016.
 */
class BootCompleteReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {

        val h = Handler()
        h.postDelayed({
            App.get(context).startService(GpsLocationService.intent(context))
        }, 10000)
    }
}
